File: _static/samples/sample.dart
import 'package:face_library/face_library.dart';

void main() {
  // [face_detection]
  FaceLibrary.loadModel(modelPath, FaceModel.faceDetector4B, ProcessingUnit.cpu);

  // Create a new instance of the FaceDetector class.
  var faceDetector = FaceDetector()
    ..confidenceThreshold = 50
    ..model = FaceModel.faceDetector4B
    ..nmsIouThreshold = 40
    ..threadCount = 4;

  // Load an image from a file.
  Image image = Image.fromFile("image1.jpg", PixelFormat.bgr24Bits);

  // Detect faces on the image.
  DetectedFaceList detectedFaceList = faceDetector.detectFaces(image);

  // Enumerate detected faces.
  for (DetectedFace face in detectedFaceList) {
    // ...
  }

  // Disposes all resources allocated to the FaceDetector.
  faceDetector.dispose();
  // [face_detection]
}

void checkLicense() {
  // [license_check]
  try {
      FaceLibrary.checkLicense("id3Face.lic");
  }
  catch (FaceException ex)
  {
      print(ex.message);
  }
  // [license_check]
}

void activateLicense() {
  // [license_activation]
  String hardwareCode = License.getHostHardwareCode(LicenseHardwareCodeType.windowsOs);

  try {
    License.activateSerialKey(hardwareCode, "XXXX-XXXX-XXXX-XXXX", "[Computer name]", "id3Face.lic");
  } catch (FaceException ex) {
    print(ex.message);
  }
  // [license_activation]
}

void FaceComparisonOnCard() {
  // [face_comparison_on_card]
  var faceDetector = FaceDetector();
  var faceEncoder = FaceEncoder();

  // Load an image from a file.
  var image = Image.fromFile("image1.jpg", PixelFormat.bgr24Bits);

  // Detect faces on the image.
  var detectedFaceList = faceDetector.detectFaces(image);

  // Create the template from the largest detected faces.
  var faceTemplate = faceEncoder.createTemplate(image, detectedFaceList.getLargestFace());

  // Export the face template object to a Biometric Data Template (BDT) buffer.
  var bdt = faceTemplate.toBdt();

  // Send BDT buffer to card
  // ...

  // Dispose all resources
  detectedFaceList.dispose();
  faceEncoder.dispose();
  faceDetector.dispose();
  // [face_comparison_on_card]
}

File: _static/samples/sample.py
import face_library

def main():

    # [face_detection]
    face_library.load_model(model_path, face_model=face_library.FaceModel.FACE_DETECTOR_4B, processing_unit=face_library.ProcessingUnit.CPU)

    # Create a new instance of the FaceDetector class.
    face_detector = face_library.FaceDetector()
    face_detector.confidence_threshold = 50
    face_detector.model = face_library.FaceModel.FACE_DETECTOR_4B
    face_detector.nms_iou_threshold = 40
    face_detector.thread_count = 4

    # Load an image from a file.
    image = face_library.Image.from_file("image1.jpg", pixel_format=face_library.PixelFormat.BGR_24_BITS)

    # Detect faces on the image.
    detected_face_list = face_detector.detect_faces(image)

    # Enumerate detected faces.
    for face in detected_face_list:
        # ...

    # Dispose of all resources allocated to the FaceDetector.
    face_detector.dispose()
    # [face_detection]

    # [face_comparison_on_card]
    face_detector = FaceDetector()
    face_encoder = FaceEncoder()

    # Load an image from a file.
    image = Image.from_file("image1.jpg", PixelFormat.BGR_24_BITS)

    # Detect faces on the image.
    detected_face_list = face_detector.detect_faces(image)

    # Create the template from the largest detected face.
    face_template = face_encoder.create_template(image, detected_face_list.largest_face)

    # Export the face template object to a Biometric Data Template (BDT) buffer.
    bdt = face_template.to_bdt()

    # Send BDT buffer to card
    # ...

    # Dispose all resources
    detected_face_list.dispose()
    face_encoder.dispose()
    face_detector.dispose()
    # [face_comparison_on_card]

if __name__ == "__main__":
    main()

